docs/map.rst

```diff
@@ -0,0 +1,602 @@
+Maps
+====
+
+.. highlight:: haskell
+
+Maps (sometimes referred to as dictionaries in other languages) allow you to
+store associations between *unique keys* and *values*. There are three
+implementations provided by the ``containers`` package:
+:haddock:`containers/Data.Map.Strict`, :haddock:`Data.Map.Lazy`, and
+:haddock:`Data.IntMap`. You almost never want the lazy version so use
+``Data.Map.Strict``, or if your keys are ``Int`` use ``Data.IntMap``; all of
+these implementations are *immutable*.
+
+::
+
+    data Map k v = ...
+
+    data IntMap v = ...
+
+.. IMPORTANT::
+   ``Map`` relies on the key type ``k`` having instances of the ``Eq`` and
+   ``Ord`` typeclass for its internal representation. These are already defined
+   for builtin types, and if you are using your own data type you can use the
+   `deriving
+   <https://en.wikibooks.org/wiki/Haskell/Classes_and_types#Deriving>`_
+   mechanism.
+
+
+Short Example
+-------------
+
+The following GHCi session shows some of the basic map functionality::
+
+    import qualified Data.Map.Strict as Map
+
+    let nums = Map.fromList [(1,"one"), (2,"two"), (3,"three")]
+
+    -- Check if we know the english words for 1 and 4
+    Map.member nums 1
+    > True
```

Comment:
Since I don't think `member` is important enough to discuss, I don't really think you should use it in examples either. Just skip it.

Reply:
Removed.